test/ex_doc/cli_test.exs

```diff
@@ -67,6 +68,70 @@ defmodule ExDoc.CLITest do
     assert io == "ExDoc v#{ExDoc.version()}\n"
   end
 
+  describe "--warnings-as-errors" do
+    @describetag :warnings
+
+    test "exits with code 0 when no warnings" do
+      ExDoc.Utils.unset_warned()
+
+      {[html, epub], _io} = run(["ExDoc", "1.2.3", @ebin, "--warnings-as-errors"])
+
+      assert html ==
+               {"ExDoc", "1.2.3",
+                [
+                  formatter: "html",
+                  formatters: ["html", "epub"],
+                  apps: [:ex_doc],
+                  source_beam: @ebin,
+                  warnings_as_errors: true
+                ]}
+
+      assert epub ==
+               {"ExDoc", "1.2.3",
+                [
+                  formatter: "epub",
+                  formatters: ["html", "epub"],
+                  apps: [:ex_doc],
+                  source_beam: @ebin,
+                  warnings_as_errors: true
+                ]}
+    end
+
+    test "exits with 1 when there is a warning" do
+      ExDoc.Utils.set_warned()
```

Comment:
as mentioned above there is a bug that this test is not catching. what about making it more realistic, have an actual .ex or .md file, and assert on the warning message?